### PR TITLE
API for review actions

### PR DESCRIPTION
**Pull request by washort**
Expose the reviewer actions as API resources.

### CODE REVIEW

File: mkt/reviewers/utils.py [hunk 1]
```diff
@@ -31,15 +31,24 @@
...
 
+def get_review_type(request, addon, version):
+    if EscalationQueue.objects.filter(addon=addon).exists():
+        queue = 'escalated'
+    elif RereviewQueue.objects.filter(addon=addon).exists():
+        queue = 'rereview'
+    else:
+        queue = 'pending'
+    return queue
+
```
Comment (eviljeff) [inline mkt/reviewers/utils.py]: I see you've just moved this from ReviewBase, but how are apps in the updates queue handled?

Comment (washort) [inline mkt/reviewers/utils.py]: as a search for approved apps with unapproved versions.

**Comment by eviljeff** [comment]
Is there a related bug for this?

**Comment by washort** [comment]
no bugs filed for it currently.

File: mkt/reviewers/views.py [hunk 2]
```diff
@@ -995,6 +996,69 @@ def post(self, request, pk, region, *args, **kwargs):
         return Response({'approved': bool(form.cleaned_data['approve'])})
 
 
+class _AppAction(SlugOrIdMixin):
+    permission_classes = [GroupPermission('Apps', 'Review')]
```
Comment (eviljeff) [inline mkt/reviewers/views.py]: AppDisable action, and AppEscalate (only when clearing/delete()) action need Apps:Edit permission. process_disable() does check this, but process_clear_escalation() doesn't currently.

File: mkt/reviewers/views.py [hunk 3]
```diff
@@ -995,6 +996,69 @@ def post(self, request, pk, region, *args, **kwargs):
...
+
+class AppApprove(_AppAction, CreateAPIView):
+    def post(self, request, pk, *a, **kw):
+        app = self.get_object()
+        handler = ReviewApp(request, app, app.latest_version, ())
+        handler.set_data(request.DATA)
+        result = handler.process_approve()
+        if result is None:
+            return Response(status=409)
+        return Response({'score': result})
```
Comment (eviljeff) [inline mkt/reviewers/views.py]: reject and clear_escalation return points scores too.

**Comment by washort** [comment]
updated

**Comment by eviljeff** [comment]
r+

**Pull request merged**